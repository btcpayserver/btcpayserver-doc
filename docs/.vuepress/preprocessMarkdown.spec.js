const { resolve } = require('path')
const preprocess = require('./preprocessMarkdown')

const pp = preprocess.bind({
  resourcePath: resolve(__dirname, '../Docker/README.md')
})

const externalFrontmatter = `---\nexternalRepo: https://github.com/btcpayserver/btcpayserver-docker\n---`

describe('preprocessMarkdown', () => {
  it('replaces YouTube links', () => {
    const md = `
Take a look at how BTCPay works in a video below.
[![](https://raw.githubusercontent.com/btcpayserver/btcpayserver-doc/master/img/thumbnails/HowBTCPayServerWorks.png)](https://www.youtube.com/watch?v=nr0UNbz3AoQ)

Here is a presentation of the global architecture at Advancing Bitcoin conference.

Short link:
[![What is a pull payment](https://img.youtube.com/vi/-e8lPd9NtPs/mqdefault.jpg)](https://youtu.be/-e8lPd9NtPs "What is a pull payment")

Other link with different title:
[![What is a pull payment](https://img.youtube.com/vi/-e8lPd9NtPs/mqdefault.jpg)](https://youtu.be/-e8lPd9NtPs 'What is a "pull payment"')`

    expect(pp(md)).toMatchSnapshot()
  })

  it('replaces external links to docs with internal VuePress links', () => {
    const md =
      externalFrontmatter +
      `
While [our instructions](https://docs.btcpayserver.org/deployment/lunanodewebdeployment) cover how to install BTCPayServer in one click on Azure or Lunanode, BTCPay Server is not limited to those options.`

    expect(pp(md)).toMatchSnapshot()
  })

  it('replaces links to docs with internal VuePress links', () => {
    const md =
      externalFrontmatter +
      `
([See the documentation](docs/pihole.md))`

    expect(pp(md)).toMatchSnapshot()
  })

  it('replaces repo internal links with external links', () => {
    const md =
      externalFrontmatter +
      `
When you run \`btcpay-setup.sh\`, your environment variables are used by [build.sh](build.sh) (or [build.ps1](build.ps1)) to generate a docker-compose adapted for your needs.
For the full list of options, see: [Environment variables](#environment-variables)

By default, the generated file is \`Generated/docker-compose.generated.yml\`, constructed from the relevant [Docker fragments](docker-compose-generator/docker-fragments) for your setup.

Available \`BTCPAYGEN_ADDITIONAL_FRAGMENTS\` currently are:

* [opt-save-storage](docker-compose-generator/docker-fragments/opt-save-storage.yml) will keep around 1 year of blocks (prune BTC for 100 GB)
* [opt-add-pihole](docker-compose-generator/docker-fragments/opt-add-pihole.yml) ([See the documentation](docs/pihole.md))`

    expect(pp(md)).toMatchSnapshot()
  })

  it('replaces external repo links with internal links', () => {
    const md =
      externalFrontmatter +
      `
[opt-unsafe-expose](https://github.com/btcpayserver/btcpayserver-docker/blob/master/README.md#generated-docker-compose)`

    expect(pp(md)).toMatchSnapshot()
  })
})
