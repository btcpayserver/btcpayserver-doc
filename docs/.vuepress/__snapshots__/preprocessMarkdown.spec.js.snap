// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preprocessMarkdown replaces YouTube links 1`] = `
"
Take a look at how BTCPay works in a video below.

<a href="https://www.youtube.com/watch?v=nr0UNbz3AoQ" title="YouTube" class="ytEmbed" data-id="nr0UNbz3AoQ" style="background-image:url(https://img.youtube.com/vi/nr0UNbz3AoQ/hqdefault.jpg);">
  <iframe
    title="YouTube"
    data-src="https://www.youtube-nocookie.com/embed/nr0UNbz3AoQ?&autoplay=1&autohide=1&modestbranding=1&color=white&rel=0"
    frameborder="0"
    allow="autoplay;encrypted-media;picture-in-picture"
    allowfullscreen
  ></iframe>
</a>

Here is a presentation of the global architecture at Advancing Bitcoin conference.

Short link:

<a href="https://youtu.be/-e8lPd9NtPs" title="What is a pull payment" class="ytEmbed" data-id="-e8lPd9NtPs" style="background-image:url(https://img.youtube.com/vi/-e8lPd9NtPs/hqdefault.jpg);">
  <iframe
    title="What is a pull payment"
    data-src="https://www.youtube-nocookie.com/embed/-e8lPd9NtPs?&autoplay=1&autohide=1&modestbranding=1&color=white&rel=0"
    frameborder="0"
    allow="autoplay;encrypted-media;picture-in-picture"
    allowfullscreen
  ></iframe>
</a>

Other link with different title:

<a href="https://youtu.be/-e8lPd9NtPs" title="What is a &quot;pull payment&quot;" class="ytEmbed" data-id="-e8lPd9NtPs" style="background-image:url(https://img.youtube.com/vi/-e8lPd9NtPs/hqdefault.jpg);">
  <iframe
    title="What is a &quot;pull payment&quot;"
    data-src="https://www.youtube-nocookie.com/embed/-e8lPd9NtPs?&autoplay=1&autohide=1&modestbranding=1&color=white&rel=0"
    frameborder="0"
    allow="autoplay;encrypted-media;picture-in-picture"
    allowfullscreen
  ></iframe>
</a>"
`;

exports[`preprocessMarkdown replaces external links to docs with internal VuePress links 1`] = `
"---
externalRepo: https://github.com/btcpayserver/btcpayserver-docker
---
While [our instructions](/deployment/lunanodewebdeployment) cover how to install BTCPayServer in one click on Azure or Lunanode, BTCPay Server is not limited to those options."
`;

exports[`preprocessMarkdown replaces external repo links with internal links 1`] = `
"---
externalRepo: https://github.com/btcpayserver/btcpayserver-docker
---
[opt-unsafe-expose](/Docker/README.md#generated-docker-compose)"
`;

exports[`preprocessMarkdown replaces links to docs with internal VuePress links 1`] = `
"---
externalRepo: https://github.com/btcpayserver/btcpayserver-docker
---
([See the documentation](/Docker/pihole.md))"
`;

exports[`preprocessMarkdown replaces repo internal links with external links 1`] = `
"---
externalRepo: https://github.com/btcpayserver/btcpayserver-docker
---
When you run \`btcpay-setup.sh\`, your environment variables are used by [build.sh](https://github.com/btcpayserver/btcpayserver-docker/blob/master/build.sh) (or [build.ps1](https://github.com/btcpayserver/btcpayserver-docker/blob/master/build.ps1)) to generate a docker-compose adapted for your needs.
For the full list of options, see: [Environment variables](#environment-variables)

By default, the generated file is \`Generated/docker-compose.generated.yml\`, constructed from the relevant [Docker fragments](https://github.com/btcpayserver/btcpayserver-docker/blob/master/docker-compose-generator/docker-fragments) for your setup.

Available \`BTCPAYGEN_ADDITIONAL_FRAGMENTS\` currently are:

* [opt-save-storage](https://github.com/btcpayserver/btcpayserver-docker/blob/master/docker-compose-generator/docker-fragments/opt-save-storage.yml) will keep around 1 year of blocks (prune BTC for 100 GB)
* [opt-add-pihole](https://github.com/btcpayserver/btcpayserver-docker/blob/master/docker-compose-generator/docker-fragments/opt-add-pihole.yml) ([See the documentation](/Docker/pihole.md))"
`;
